### PR TITLE
fix slice and sqltable relationship - datasource type should not be null

### PR DESCRIPTION
**Pull request by ofekisr**
#14482 

After the fix inside Slice model, datasource_type column cannot be null

### CODE REVIEW

File: superset/migrations/versions/1a51759ec9f2_fix_slices_datasource_type.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+down_revision = '67da9ef1ef9c'
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import mysql
+
+
+def upgrade():
+    op.alter_column('slices', 'datasource_type',
+                    existing_type=mysql.VARCHAR(length=200),
```
Comment (villebro) [inline superset/migrations/versions/1a51759ec9f2_fix_slices_datasource_type.py:36]: We support other databases than MySQL for the metadata database. This should be the generic String type, i.e. `sqlalchemy.String(200)`

File: superset/migrations/versions/1a51759ec9f2_fix_slices_datasource_type.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                    nullable=False)
+
+
+def downgrade():
+    op.alter_column('slices', 'datasource_type',
+                    existing_type=mysql.VARCHAR(length=200),
```
Comment (villebro) [inline superset/migrations/versions/1a51759ec9f2_fix_slices_datasource_type.py:42]: Same here

**Review by villebro — CHANGES_REQUESTED** [review]
Left a few comments. Also, there have been many migrations lately, so make sure you're on the correct head.

**Review by ofekisr — COMMENTED** [review]
@villebro 
the change from VARCHAR to another type should be in separate PR.
in that PR I haven't changed the type.

**Pull request closed without merging**